source/common/http/conn_manager_impl.cc

```diff
@@ -1741,7 +1744,6 @@ void ConnectionManagerImpl::ActiveStreamFilterBase::commonContinue() {
 
   ENVOY_STREAM_LOG(trace, "continuing filter chain: filter={}", parent_,
                    static_cast<const void*>(this));
-  ASSERT(!canIterate());
```

Comment:
I really like this solution! It's simple and it works! The only concern I have is the removal of ASSERT here and below. Is it possible to keep the assert? For example: ASSERT(!canIterate() || inline_buffered_data_)?

Reply:
yep that makes sense. I was ripping out all the asserts that failed, makes sense to make it more exact instead